Allow for skip to be defined on a group-level skipping all checks inside

Wicked7000 opened the pull request:
This fixes issue #649 allowing groups to be defined to be skipped. this is done by setting 'Skip' on the group to true. When checks are run this will then set the state of any checks within that group to info (the check is still technically run but just overridden to be INFO as I wasn't sure on what the desired behaviour here was, we could just skip the check entirely and not run and just set it to INFO).

Please let me know if there are any additional changes you want done to this!

yoavrotems commented:
Hey first of all thanks for contributing, secondly your doubt was right If a test is skip we don't want to run it at all. 
And I think instead of adding new code the handle skip for group and change the state to INFO as you did maybe we should just inherit it the the group tests? Like if we have a group marked as skip then we could add the skip to every test inside the group so we will still handle it correctly and take care of the test as needed for example https://github.com/aquasecurity/kube-bench/blob/master/check/check.go#L113 where we add the right reason for the test state.
WDYT ?

Wicked7000 commented:
> Hey first of all thanks for contributing, secondly your doubt was right If a test is skip we don't want to run it at all.
> And I think instead of adding new code the handle skip for group and change the state to INFO as you did maybe we should just inherit it into the group tests? Like if we have a group marked as skip then we could add the skip to every test inside the group so we will still handle it correctly and take care of each test as needed for example https://github.com/aquasecurity/kube-bench/blob/master/check/check.go#L113 where we add the right reason for the test state.
> WDYT ?

Yeah apologies that makes sense, I've changed the code to just switch over the type to skip and adjusted the tests for that. (it uses a normal runner rather than mocking it otherwise I don't think it would really be testing anything, but I don't think that should be a problem as its not running the actual test code anyways)

Let me know if you think anything else needs to be changed!

yoavrotems reviewed (approved):
LGTM, really good job and thanks for contributing 👍
